### DISCUSSION TITLE
200区块之后时间未到

### DISCUSSION THREAD

**Discussion by a2asister** [openbuildxyz/mina-zkapp-bootcamp#374] [Q&A]
源文件：https://github.com/missop/mina-zkapp-bootcamp/blob/task4/learn/missop/task4/FundMe.ts   

现象：执行unit/FundMe.test.ts时第二次执行 payout 方法报错，报错如下：

  ● FundMe › should payout successfully with right preimage after deadline 200 blocks

    Transaction failed with errors:
    - [[],[["Cancelled"]],[["Update_not_permitted_timing"]]]

      116 |         });
      117 |         await tx2.prove();
    > 118 |         await tx2.sign([deployer.key]).send();
          |         ^
      119 |         expect(funeMeContract.account.balance.get()).toEqual(allBalance.mul(7).div(10));
      120 |
      121 |         local.setBlockchainLength(endTime.add(405));

      at node_modules/o1js/src/lib/mina/transaction.ts:528:17
      at Object.<anonymous> (src/FundMe/unit/FundMe.test.ts:118:9)

**Comment by tokenlin** [comment]
第一次调用时用payout()，后面不能再重复调用。因为已经转给用户，token的锁定时间不在合约里面计算，是单独计算。

另外，用local.incrementGlobalSlot(200);表示通过的时间，如下参考。

代码参考来自“https://github.com/coldstar1993/mina-sz/blob/main/contracts/src/others/simple-zkapp-vesting.ts”

```
        // local.setBlockchainLength(endTime.add(203));
        local.incrementGlobalSlot(200);
        let anotherGuyAddr = PrivateKey.random().toPublicKey();
        const tx2 = await Mina.transaction(deployer, async () => {
            AccountUpdate.fundNewAccount(deployer);// 消耗1MINA用于创建新账户
            const acctUpdate = AccountUpdate.createSigned(deployer);
            acctUpdate.send({ to: anotherGuyAddr, amount: 9e9 });
        });
        await tx2.sign([deployer.key]).send();
```

**Reply by a2asister** [reply]
let anotherGuyAddr = PrivateKey.random().toPublicKey();
        const tx2 = await Mina.transaction(deployer, async () => {
            AccountUpdate.fundNewAccount(deployer);// 消耗1MINA用于创建新账户
            const acctUpdate = AccountUpdate.createSigned(deployer);
            acctUpdate.send({ to: anotherGuyAddr, amount: 9e9 });
        });
这一块是把deployer的资金释放给了anotherGuyAddr，而不是把合约的资金给释放了吧
